### DISCUSSION TITLE
on tugger and rusty_jsc

### DISCUSSION THREAD

**Discussion by grue-bleen** [tauri-apps/tauri#9593] [Ideas]
tauri-bundler vs. tugger:

[tugger](https://pyoxidizer.readthedocs.io/en/stable/tugger_overview.html) and is a rust crate belonging to the [pyoxidizer](https://pyoxidizer.readthedocs.io/) project.  the tauri project could benefit by integrating tugger and inheriting the broader feature set.

javascriptcore vs. rusty_jsc

[rusty_jsc](https://github.com/wasmerio/rusty_jsc) is similarly a part of the [wasmer](https://wasmer.io/) runtime, and the recent addition of javascriptcore into wasmer comes with significant [performance gains](https://wasmer.io/posts/wasmer-3_3-and-javascriptcore).

wasmer, tauri, and pyoxidizer would all benefit from finding better reuse and alignment across their rust crates.  perhaps this then paves the way towards an in-process extension model with python, wasm, and undoubtedly others?

**Comment by FabianLars** [comment]
tugger was on my list for a long time for their rust-y apple code signing implementation. Bundle types wise they don't support anything more than we do so there's not much to gain on our side. For them they may be better off collaborating with cargo-packager (or similar).

jsc_rusty alone is irrelevant for us because we don't use jack directly, it's just dependency of webkitgtk.

In combination with wasmer though it's indeed more interesting but currently we don't do/have anything wasm/wasi in tauri so I don't see any collaboration here either anytime soon.

Thanks for starting this discussion btw!
